lib/ruby_lsp/requests/base_request.rb

```diff
@@ -30,6 +30,51 @@ def range_from_syntax_tree_node(node)
           end: LanguageServer::Protocol::Interface::Position.new(line: loc.end_line - 1, character: loc.end_column),
         )
       end
+
+      sig { params(node: SyntaxTree::ConstPathRef).returns(String) }
+      def full_constant_name(node)
+        name = +node.constant.value
+        constant = T.let(node, SyntaxTree::Node)
+
+        while constant.is_a?(SyntaxTree::ConstPathRef)
+          constant = constant.parent
+
+          case constant
+          when SyntaxTree::ConstPathRef
+            name.prepend("#{constant.constant.value}::")
+          when SyntaxTree::VarRef
+            name.prepend("#{constant.value.value}::")
+          end
+        end
+
+        name
+      end
+
+      sig do
+        params(
+          parent: SyntaxTree::Node,
+          target_nodes: T::Array[T.class_of(SyntaxTree::Node)],
+          position: Integer,
+        ).returns(T::Array[SyntaxTree::Node])
+      end
+      def locate_node_and_parent(parent, target_nodes, position)
```

Comment:
@paracycle @kddnewton @vinistock Following the earlier discussion around node-locating API: In the current use cases we have in `ruby-lsp`, I'd expect that API to:

- Locate certain types of nodes with a given position.
- Return both the node and its parent. Having the parent allows us to locate nodes (like `(vcall (ident "foo"))`) based on a child node (like `(ident "foo")`). We use this in `documentHighlight`.